src/data/list/forall2.lean

```diff
@@ -167,6 +167,17 @@ lemma rel_append : (forall₂ r ⇒ forall₂ r ⇒ forall₂ r) append append
 | [] [] h l₁ l₂ hl := hl
 | (a::as) (b::bs) (forall₂.cons h₁ h₂) l₁ l₂ hl := forall₂.cons h₁ (rel_append h₂ hl)
 
+lemma rel_reverse : (forall₂ r ⇒ forall₂ r) reverse reverse
+| [] [] forall₂.nil := forall₂.nil
+| (a::as) (b::bs) (forall₂.cons h₁ h₂) :=
+  by simp only [reverse_cons];
+  from rel_append (rel_reverse h₂) (forall₂.cons h₁ forall₂.nil)
+
+lemma forall₂_iff_reverse {l₁ l₂} : forall₂ r l₁ l₂ ↔ forall₂ r l₁.reverse l₂.reverse :=
+iff.intro
+  (assume h, rel_reverse h)
+  (assume h, by rw [← reverse_reverse l₁, ← reverse_reverse l₂]; from rel_reverse h)
```

Comment:
If you reverse this it makes a reasonable simp lemma:
```suggestion
@[simp]
lemma forall₂_iff_reverse {l₁ l₂} : forall₂ r l₁.reverse l₂.reverse ↔ forall₂ r l₁ l₂ :=
iff.intro
  (assume h, by rw [← reverse_reverse l₁, ← reverse_reverse l₂]; from rel_reverse h)
  (assume h, rel_reverse h)
```

Reply:
Looks good to me, thanks for the suggestion!